prism/src/main/java/org/apache/falcon/resource/extensions/ExtensionManager.java

```diff
@@ -292,39 +296,76 @@ public APIResult delete(@PathParam("job-name") String jobName,
 
     @POST
     @Path("submit/{extension-name}")
-    @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN})
+    @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.MULTIPART_FORM_DATA})
     @Produces({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON})
     public APIResult submit(
             @PathParam("extension-name") String extensionName,
             @Context HttpServletRequest request,
-            @DefaultValue("") @QueryParam("doAs") String doAsUser) {
+            @DefaultValue("") @QueryParam("doAs") String doAsUser,
+            @QueryParam("jobName") String jobName,
+            @FormDataParam("entities") List<Entity> entities,
+            @FormDataParam("config") InputStream config) {
         checkIfExtensionServiceIsEnabled();
         try {
-            List<Entity> entities = generateEntities(extensionName, request);
+            ExtensionType extensionType = getExtensionType(extensionName);
+            if (ExtensionType.TRUSTED.equals(extensionType)) {
+                entities = generateEntities(extensionName, request);
+            }
+            List<String> entityNames = new ArrayList<>();
+            validateEntities(entities);
             for (Entity entity : entities) {
                 submitInternal(entity, doAsUser);
+                entityNames.add(entity.getName());
             }
+            byte[] configBytes = IOUtils.toByteArray(config);
+            ExtensionMetaStore metaStore = ExtensionStore.get().getMetaStore();
+            metaStore.storeExtensionJob(jobName, extensionName, extensionType, entityNames, configBytes);
         } catch (FalconException | IOException e) {
             LOG.error("Error when submitting extension job: ", e);
             throw FalconWebException.newAPIException(e, Response.Status.INTERNAL_SERVER_ERROR);
         }
         return new APIResult(APIResult.Status.SUCCEEDED, "Extension job submitted successfully");
     }
 
+    private void validateEntities(List<Entity> entities) throws FalconException {
+        for (Entity entity : entities) {
+            super.validate(entity);
+        }
+    }
+
+    private ExtensionType getExtensionType(String extensionName) {
+        ExtensionMetaStore metaStore = ExtensionStore.get().getMetaStore();
+        ExtensionBean extensionDetails = metaStore.getDetail(extensionName);
+        return extensionDetails.getExtensionType();
+    }
+
     @POST
     @Path("submitAndSchedule/{extension-name}")
-    @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN})
+    @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.MULTIPART_FORM_DATA})
     @Produces({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON})
     public APIResult submitAndSchedule(
```

Comment:
Possible to have a common submit method that both submit and submitAndSchedule can use?